api/src/license_operator.rs

```diff
@@ -0,0 +1,207 @@
+use std::borrow::Borrow;
+
+use log::info;
+use rand::rngs::ThreadRng;
+use rand::Rng;
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+use crate::adler32::adler32_checksum;
+use crate::license_key::LicenseKeyStatus;
+use crate::license_magic::LicenseMagic;
+use crate::license_serializer::{DefaultLicenseKeySerializer, LicenseKeySerializer};
+use crate::LicenseKey;
+
+pub struct LicenseOperator {
+    rng: ThreadRng,
+
+    magic: LicenseMagic,
+    key_size: usize,
+    checksum_init: (u32, u32),
+
+    serializer: Box<dyn LicenseKeySerializer>,
+
+    blacklist: Vec<Vec<u8>>,
+    byte_check: Vec<(usize, LicenseMagic)>,
+}
+
+impl LicenseOperator {
+    #[inline(always)]
+    pub fn new(
+        magic: LicenseMagic,
+        key_size: usize,
+        checksum_init: (u32, u32),
+        serializer: Box<dyn LicenseKeySerializer>,
+        blacklist: Vec<Vec<u8>>,
+        byte_check: Vec<(usize, LicenseMagic)>,
+    ) -> LicenseOperator {
+        LicenseOperator {
+            rng: rand::thread_rng(),
+            magic,
+            key_size,
+            checksum_init,
+            serializer,
+            blacklist,
+            byte_check,
+        }
+    }
+
+    #[inline(always)]
+    pub fn default() -> LicenseOperator {
+        LicenseOperator {
```

Comment:
You can use
```rust
Self {
    rng: rand::thread_rng(),
    magic: ...,
    ...
}
```

Self is type alias for current type

Reply:
hmmm... I will use it in next push.